src/backend/utils/page-builder.js

```diff
@@ -0,0 +1,264 @@
+ *       columns: 3
+ *     })
+ *     this.addInfoBlock({
+ *       title: 'The number of users',
+ *       value: usersCount,
+ *       icon: 'fas fa-star fa-2x',
+ *       columns: 3
+ *     })
+ *     return {
+ *       title: this.title,
+ *       subtitle: this.subtitle,
+ *       content: this.convertedPageContent()
+ *     }
+ *   }
+ * }
+ *
+ */
+class PageBuilder {
+  /**
+   *
+   * @param  {AdminBro}     options.admin  current instance of AdminBro
+   */
+  constructor({ admin }) {
+    this._admin = admin
+
+    /**
+     * List of HTML elements as JavaScript string
+     * @type {String[] | Function | null}
+     */
+    this._pageContent = []
+    this.title = null
+    this.subtitle = null
+    this.charts = {}
+    this.types = {
+      warning: '#ff9f89',
+      danger: '#f0616f',
+      succes: '#21c197',
+      info: '#718af4',
+    }
+  }
+
+  /**
+   * Returns string of HTML elements
+   * Can be overwritten in custom pages
+   * @return {String}
+   */
+  async render() {
+    await this.build()
+    return {
+      title: this.title,
+      subtitle: this.subtitle,
+      content: this.convertedPageContent(),
+      charts: this.charts,
+    }
+  }
+
+  build() {
+    throw new NotImplementedError()
+  }
+
+  /** Returns string from joined and wrapped array of html elements
+   * @return {String}
+   */
+  convertedPageContent() {
+    if (this._pageContent.length > 0) {
+      const pageContentAsString = this._pageContent.join('')
+      return `<div class="columns is-multiline dashboard-content"> ${pageContentAsString} </div>`
+    }
+    return null
+  }
+
+  addChart(options) {
+    const { columns, offset, config } = options
+    const chart = `
+      <div class="column is-12-tablet is-${columns}-desktop is-offset-${offset || 0}"> 
+        <canvas class="chart" data-chart=${config.name}> 
+        </canvas>
+      </div>`
+    this._pageContent.push(chart)
+    this.charts[config.name] = config
+  }
+
+  addInfoList(options) {
+    const { items, columns, offset, title, subtitle } = options
+    const itemsContent = items.map(item => this.getInfoListItem(item))
+    const infoList = `
+      <div class="column is-12-tablet is-${columns}-desktop is-offset-${offset || 0}"> 
+        <div class="info-list border-box">
+          <div class="h2">
+            ${title}
+          </div>
+          <div class="info-subtitle">
+            ${subtitle}
+          </div>
+          <div class="items">
+            ${itemsContent.join('<div class="item-spacer"> </div>')}
+          </div>
+        </div>
+      </div>`
+    this._pageContent.push(infoList)
+  }
+
+  getStatusHtml(status) {
+    const statusHtml = `
+      <div class="status ${status}">
+        ${status.toUpperCase()}
+      </div>`
+    return statusHtml
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getInfoListItem(item) {
+    const { title, subtitle, content, status, date, imgSrc } = item
```

Comment:
in general you can move destructuring parameters up to function definition and also add defaults there like: title = 'Example title'